pulp_rpm/app/downloaders.py

```diff
@@ -12,18 +10,43 @@
 class RpmDownloader(HttpDownloader):
     """
     Custom Downloader that automatically handles authentication token for SLES repositories.
+
+    Args:
+        silence_errors_for_response_status_codes (iterable): An iterable of response exception
+            codes to be ignored when raising exception. e.g. `{404}`
+        sles_auth_token (str): SLES authentication token.
+
+    Raises:
+        FileNotFoundError: If aiohttp response status is 404 and silenced.
+
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, silence_errors_for_response_status_codes=set(), sles_auth_token=None,
+                 **kwargs):
         """
         Initialize the downloader.
         """
-        if 'sles_auth_token' in kwargs:
-            self.sles_auth_token = kwargs.pop('sles_auth_token')
-        else:
-            self.sles_auth_token = None
+        self.sles_auth_token = sles_auth_token
+        self.silence_errors_for_response_status_codes = silence_errors_for_response_status_codes
+
         super().__init__(*args, **kwargs)
 
+    def raise_for_status(self, response):
+        """
+        Raise error if aiohttp response status is >= 400 and not silenced.
+
+        Raises:
+            FileNotFoundError: If aiohttp response status is 404 and silenced.
+            aiohttp.ClientResponseError: If the response status is 400 or higher and not silenced.
+
+        """
+        silenced = response.status in self.silence_errors_for_response_status_codes
+
+        if silenced and response.status == 404:
+            raise FileNotFoundError()
+
+        response.raise_for_status()
```

Comment:
I think there is a logic error here. This will raise for non 404 cases, e.g. 400 even if the downloader is told to silence them.

```suggestion
        silenced = response.status in self.silence_errors_for_response_status_codes

        if not silenced:
            response.raise_for_status()
         else if response.status == 404:
            raise FileNotFoundError()
```

Reply:
the downloader was told to silence 404, not 400!
silence_errors_for_response_status_codes = {404}

Reply:
That's fair, but then the issue is that the `silence_errors_for_response_status_codes` only works if you set it equal to 404, which makes that kwarg written as if it's generalized, but it really isn't.